### PR TITLE
feat: rewrite server create to registration token-based wizard

### PR DESCRIPTION
**Pull request by jisung-02**
## Summary

Replaces the old group-based server creation flow with a 3-step interactive
wizard that matches the Alpacon frontend UX. The command now guides the user
through OS selection, server naming, and registration token selection, then
fetches and displays the Alpamon installation guide from the API.

## Interactive flow

```
Step 1 — OS selection
  What is the server's OS?
    [1] Debian family (Ubuntu, Debian)
    [2] RedHat family (RHEL, CentOS, Fedora, Amazon/Rocky/Oracle Linux)
    [3] macOS (Apple Silicon, Intel)
  OS: 1

Step 2 — Server name (optional)
  Server name (optional—hostname will be used if not specified):
  Server Name:

Step 3 — Registration token
  Select a registration token:
    [1] prod-token
    [2] dev-token
    [+] Create new token
  Token: +
  Token name: staging-token
  Warning: New token created. Save the key now—it will not be shown again: alpacax_...

Step 4 — Installation guide output
  ──────────────────── Installation guide ────────────────────
  Platform : Debian family
  URL      : https://alpacon.io

  Step 1 — Install Alpamon
  curl -fsSL https://... | sudo bash

  Step 2 — Register
  alpamon register --url https://alpacon.io --token alpacax_...

  Step 3 — Verify
  Your server will appear in the Servers list within moments.
```

## Non-interactive mode (all flags provided)

```bash
alpacon server create --platform debian --token prod-token --name my-server
alpacon server create --platform rhel --new-token staging-token
alpacon server create --platform debian --token prod-token --json
```

## Changes

**`api/server/types.go`**
- Replace `ServerRequest` / old `ServerCreatedResponse` with:
  `RegistrationTokenRequest`, `RegistrationTokenCreatedResponse` (with `Key`),
  `RegistrationTokenDetails`, `RegistrationMethodGuideRequest`,
  `RegistrationMethodGuideJsonResponse`

**`api/server/server.go`**
- Add constants: `registrationTokenURL`, `registrationGuideURL`
- Rename `CreateServer` → `CreateRegistrationToken`
- Add: `GetRegistrationTokenByName`, `ListRegistrationTokens`, `GetRegistrationGuideJSON`
- Add sentinel error `ErrRegistrationTokenNotFound`

**`cmd/server/server_create.go`**
- Full rewrite: interactive wizard + `--platform`, `--name`, `--token`,
  `--new-token`, `--json` flags
- Use `MarkFlagsMutuallyExclusive("token", "new-token")` for Cobra-native flag validation
- Handle empty token list gracefully—skip selection prompt and go directly to token creation
- Render installation guide as structured CLI output from JSON response

**`api/server/server_test.go`**
- Update `TestCreateServer` → `TestCreateRegistrationToken`
- Add `TestListRegistrationTokens`

### Universal
- [ ] Self-reviewed the code
- [ ] No hardcoded secrets or credentials

### Go
- [ ] Error handling is complete
- [ ] Tests include edge cases
- [ ] `go mod tidy` executed

### CODE REVIEW

File: cmd/server/server.go [hunk 1]
```diff
@@ -24,4 +24,5 @@ func init() {
 	ServerCmd.AddCommand(serverCreateCmd)
 	ServerCmd.AddCommand(serverDeleteCmd)
 	ServerCmd.AddCommand(serverUpdateCmd)
+	ServerCmd.AddCommand(serverTokenCmd)
 }
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/server/server.go:27]: Fixed in 783909b — added `'alpacon server token'` to the subcommand error hint.

File: cmd/server/server_token.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+var serverTokenRegenerateCmd = &cobra.Command{
+	Use:     "regenerate SERVER",
+	Aliases: []string{"regen"},
+	Short:   "Regenerate a registration token for a server",
+	Long: `
+	Regenerate the registration token for the given server.
+	The old token is revoked and a new one is issued. The new token is shown only once—save it immediately.
+	`,
+	Example: `alpacon server token regenerate my-server`,
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/server/server_token.go]: Fixed in 783909b — renamed the positional arg to `TOKEN` and updated Short/Long/Example accordingly.

File: cmd/server/server_token.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	Args:    cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		serverName := args[0]
+
+		alpaconClient, err := client.NewAlpaconAPIClient()
+		if err != nil {
+			utils.CliErrorWithExit("Connection to Alpacon API failed: %s. Consider re-logging.", err)
+		}
+
+		response, err := server.RegenerateRegistrationToken(alpaconClient, serverName)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/server/server_token.go]: Fixed in 783909b — renamed `serverName` → `tokenName` and applied the suggested diff.

File: cmd/server/server_token.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+func printRegistrationTokenInfo(response server.ServerCreatedResponse) {
+	fmt.Fprintln(os.Stderr)
+	utils.PrintHeader("Server registration token created")
+	fmt.Fprintf(os.Stderr, "Name:  %s\n", response.Name)
+	fmt.Fprintln(os.Stderr)
+	fmt.Fprintln(os.Stderr, "Registration token (shown only once—save it now):")
+	fmt.Fprintln(os.Stderr, utils.Green(response.Key))
+	fmt.Fprintln(os.Stderr)
+	utils.CliWarning("Use this token to register an Alpamon agent. After leaving this page, you cannot retrieve the token again.")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/server/server_token.go]: Fixed in 783909b — changed the header to "Registration token regenerated" and updated the warning to say "After exiting" instead of "After leaving this page".

File: cmd/server/server_create.go [hunk 5]
```diff
@@ -5,109 +5,174 @@ import (
...
 	Use:   "create",
-	Short: "Create a new server",
+	Short: "Register a new server with a registration token",
 	Long: `
-	Create a new server with specific configurations. This command allows you to set up a server with a unique name, 
-	choose a platform, and define access permissions for different groups. 
+	Register a new server by selecting a registration token and generating an installation guide.
+	The guide includes the alpamon register command to run on your server.
+
+	When --platform and --token are provided, the command runs non-interactively.
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/server/server_create.go]: Fixed in 783909b — updated the Long description to read: "When --platform and either --token or --new-token are provided, the command runs non-interactively."

File: api/server/server.go [hunk 6]
```diff
@@ -137,3 +139,69 @@ func CreateServer(ac *client.AlpaconClient, serverRequest ServerRequest) (Server
...
+	}
+
+	if err = DeleteRegistrationToken(ac, existing.ID); err != nil {
+		return ServerCreatedResponse{}, err
+	}
+
+	return CreateRegistrationToken(ac, RegistrationTokenRequest{
+		Name:          existing.Name,
+		AllowedGroups: existing.AllowedGroups,
+	})
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/server/server.go]: Fixed in 783909b — adopted the suggested approach: `CreateRegistrationToken` is now called first; `DeleteRegistrationToken` only runs after a successful create. If delete fails, the newly created token is rolled back (deleted) before returning the error, preventing token duplication.

File: api/server/server.go [hunk 7]
```diff
@@ -137,3 +139,69 @@ func CreateServer(ac *client.AlpaconClient, serverRequest ServerRequest) (Server
...
+}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/server/server.go]: Fixed in 783909b — added `TestRegenerateRegistrationToken` which asserts that `CreateRegistrationToken` is called before `DeleteRegistrationToken` and verifies the returned token ID and key.

File: cmd/server/server_create.go [hunk 8]
```diff
@@ -1,113 +1,191 @@
...
-		platform := utils.PromptForInput("Platform (debian, rhel): ")
-		if strings.ToLower(platform) == "debian" || strings.ToLower(platform) == "rhel" {
-			return platform
+		input := strings.TrimSpace(utils.PromptForRequiredInput("OS: "))
+		idx := utils.SplitAndParseInt(input)
+		if len(idx) == 1 && idx[0] >= 1 && idx[0] <= len(platforms) {
+			return platforms[idx[0]-1].value
 		}
-		utils.CliWarning("Invalid platform. Please choose 'debian' or 'rhel'.")
+		utils.CliWarning("Invalid selection. Enter a number between 1 and %d.", len(platforms))
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/server/server_create.go]: Fixed. Reverted the platform selection from a numbered menu back to text input (matching the original implementation), and added `darwin` as a valid option alongside `debian` and `rhel`. This removes the `SplitAndParseInt` call entirely, so any non-numeric input no longer exits the process—the loop now re-prompts correctly with an error message.

File: cmd/server/server_create.go [hunk 9]
```diff
@@ -1,113 +1,191 @@
...
-		groupID, err := iam.GetGroupIDByName(ac, groupList[groupIndex-1].Name)
-		if err != nil {
-			utils.CliErrorWithExit("Group '%s' not found. Please verify the group exists and try again", groupList[groupIndex-1].Name)
+		idx := utils.SplitAndParseInt(input)
+		if len(idx) == 1 && idx[0] >= 1 && idx[0] <= len(tokens) {
+			return tokens[idx[0]-1].ID
 		}
-
-		groupIDs = append(groupIDs, groupID)
+		utils.CliWarning("Invalid selection. Enter a number from the list or '+' to create a new token.")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/server/server_create.go:176]: Fixed. Replaced `utils.SplitAndParseInt` with `strconv.Atoi` directly. On parse failure the error is discarded and execution falls through to the `CliWarning` re-prompt, so any non-numeric input (other than `+`) now loops correctly instead of exiting the process.

File: api/server/server.go [hunk 10]
```diff
@@ -123,17 +128,91 @@ func UpdateServer(ac *client.AlpaconClient, serverName string) ([]byte, error) {
...
+	}
+
+	for _, t := range response.Results {
+		if t.Name == name {
+			return t, nil
+		}
+	}
+
+	return RegistrationTokenDetails{}, ErrRegistrationTokenNotFound
+}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/server/server.go:160]: Fixed. Replaced the single-page `SendGetRequest` call with `api.FetchAllPages`, passing the same `search` param. All pages are now fetched before the exact-name match, so a token on any page will be found correctly and `ErrRegistrationTokenNotFound` is only returned when the token genuinely does not exist.

**Review by eunyoung14 — APPROVED** [review]
